### ISSUE TITLE
Composer dump returns deprecation dialog

### ISSUE THREAD

**Issue by LocalHeroPro** [PrestaShop/PrestaShop#20986]
#### Describe the bug

Something wrong with code,

#### Expected behavior

No deprecation's and similar. 

#### Steps to Reproduce

```bash-script
$ composer dump-autoload -o
```
retuens
```bash-script
Deprecation Notice: Class Tests\Core\Checkout\TermsAndConditionsTest located in ./tests/Unit/Core/Checkout/TermsAndConditionsTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Tests\Unit\PrestaShopBundle\Utils\BoolParserTest located in ./tests/Unit/Core/Util/BoolParserTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Tests\Unit\Core\Util\Country\ZipCodePatternResolverTest located in ./tests/Unit/Core/Domain/Country/ZipCodePatternResolverTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Tests\PrestaShopBundle\Routing\Converter\RoutingCacheKeyGeneratorTest located in ./tests/Unit/PrestaShopBundle/Routing/Converter/RoutingCacheKeyGeneratorTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Tests\PrestaShopBundle\Routing\Converter\LegacyUrlConverterTest located in ./tests/Unit/PrestaShopBundle/Routing/Converter/LegacyUrlConverterTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Tests\PrestaShopBundle\Routing\Converter\RouterProviderTest located in ./tests/Unit/PrestaShopBundle/Routing/Converter/RouterProviderTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Tests\PrestaShopBundle\Routing\Converter\CacheProviderTest located in ./tests/Unit/PrestaShopBundle/Routing/Converter/CacheProviderTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Tests\PrestaShopBundle\Routing\Converter\LegacyRouteTest located in ./tests/Unit/PrestaShopBundle/Routing/Converter/LegacyRouteTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Sensio\Bundle\FrameworkExtraBundle\Tests\Request\ParamConverter\ArgumentNameConverterTest located in ./vendor/sensio/framework-extra-bundle/Tests/Request/ArgumentNameConverterTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Sensio\Bundle\FrameworkExtraBundle\Tests\DependencyInjection\AddExpressionLanguageProvidersPassTest located in ./vendor/sensio/framework-extra-bundle/Tests/DependencyInjection/Compiler/AddExpressionLanguageProvidersPassTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Sensio\Bundle\FrameworkExtraBundle\Tests\DependencyInjection\AddParamConverterPassTest located in ./vendor/sensio/framework-extra-bundle/Tests/DependencyInjection/Compiler/AddParamConverterPassTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class PrestaShop\TranslationToolsBundle\Tests\Translation\Dumper\MultilanguageCatalogTest located in ./vendor/prestashop/translationtools-bundle/Tests/Translation/MultilanguageCatalogTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class PrestaShop\TranslationToolsBundle\Tests\Translation\Dumper\TraitExtractorTest located in ./vendor/prestashop/translationtools-bundle/Tests/Translation/Extractor/TraitExtractorTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class PrestaShop\TranslationToolsBundle\Tests\Translation\Extractor\FlattenizerTest located in ./vendor/prestashop/translationtools-bundle/Tests/Translation/Extractor/Util/FlattenizerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class PrestaShopBundle\Install\SimplexmlElement located in ./src/PrestaShopBundle/Install/Simplexml.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Classes\SearchCoreTest located in ./tests-legacy/Integration/classes/SearchCoreTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Integration\Classes\Db\DbTest located in ./tests-legacy/Integration/classes/db/DbTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Classes\MediaCoreTest located in ./tests-legacy/Integration/classes/MediaCoreTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Integration\ModuleGetPossibleHooksListTest located in ./tests-legacy/Integration/classes/module/ModuleGetPossibleHooksListTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Integration\ModulesGetOverrideTest located in ./tests-legacy/Integration/classes/module/ModuleGetOverrideTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Integration\ModuleOverrideInstallUninstallTest located in ./tests-legacy/Integration/classes/module/ModuleOverrideInstallUninstallTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Classes\ShopCoreTest located in ./tests-legacy/Integration/classes/ShopCoreTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Classes\CacheCoreTest located in ./tests-legacy/Unit/classes/CacheCoreTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Core\Addon\ThemeRepositoryTest located in ./tests-legacy/Unit/Core/Addon/Theme/ThemeRepositoryTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Core\Addon\ThemeValidatorTest located in ./tests-legacy/Unit/Core/Addon/Theme/ThemeValidatorTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Core\Addon\Module\ModuleManagerTest located in ./tests-legacy/Unit/Core/Addon/Module/ModuleManagerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Core\Addon\Module\ModuleRepositoryTest located in ./tests-legacy/Unit/Core/Addon/Module/ModuleRepositoryTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Stock\FakeStockAvailable4759 located in ./tests-legacy/Unit/Core/Stock/Core_Stock_StockManagerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Stock\FakeProduct4759 located in ./tests-legacy/Unit/Core/Stock/Core_Stock_StockManagerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Stock\FakePackItemsManager4759 located in ./tests-legacy/Unit/Core/Stock/Core_Stock_StockManagerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Stock\StockAvailableTest located in ./tests-legacy/Unit/Core/Stock/Core_Stock_StockManagerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Product\PriceFormatter located in ./tests-legacy/Unit/Core/Business/Product/ProductPresenterTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Product\ProductPresenterTest located in ./tests-legacy/Unit/Core/Business/Product/ProductPresenterTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Product\Search\SearchTest located in ./tests-legacy/Unit/Core/Business/Product/Search/SearchTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Product\Search\URLFragmentSerializerTest located in ./tests-legacy/Unit/Core/Business/Product/Search/URLFragmentSerializerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Product\Search\PaginationTest located in ./tests-legacy/Unit/Core/Business/Product/Search/PaginationTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Product\Search\SortOrderTest located in ./tests-legacy/Unit/Core/Business/Product/Search/SortOrderTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Payment\PaymentOptionFormDecoratorTest located in ./tests-legacy/Unit/Core/Business/Payment/PaymentOptionFormDecoratorTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Core\Localization\DataLayer\LocaleCacheDataLayerTest located in ./tests-legacy/Unit/Core/Localization/LocaleCacheDataLayerTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\Classes\Product\SpecificPrice\SpecificPriceFormatterTest located in ./tests-legacy/Unit/Classes/Product/SpecificPriceFormatterTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class LegacyTests\Unit\classes\Smarty\TemplateFinderTest located in ./tests-legacy/Unit/Classes/Smarty/TemplateFinderTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Csa\Bundle\GuzzleBundle\Tests\GuzzleHttp\Subscriber\DoctrineAdapterTest located in ./vendor/csa/guzzle-bundle/src/Tests/GuzzleHttp/Subscriber/CacheSubscriberTest.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in /usr/share/php/Composer/Autoload/ClassMapGenerator.php:201
```
**Screenshots**

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: 7.3.22-FPM

**Comment by khouloudbelguith** [comment]
Hi @LocalHeroPro,

I did not manage to reproduce the issue with 177x branch using PHP v7.3.15
![image](https://user-images.githubusercontent.com/16067358/93185035-3dcafb80-f73d-11ea-95ee-11d47bf5170f.png)

Thanks to check and feedback.

**Comment by PierreRambaud** [comment]
I manage to reproduce the issue:
Branch: develop
Composer version 1.10.1 2020-03-13 20:34:27

**Comment by LocalHeroPro** [comment]
It's weird, because in admin panel I have 1.7.6.7 version:
![image](https://user-images.githubusercontent.com/42643563/93237929-0a11c500-f781-11ea-8a98-9402afce9d36.png)
not develop branch as marked on that thread.

**Comment by PierreRambaud** [comment]
> It's weird, because in admin panel I have 1.7.6.7 version:
> ![image](https://user-images.githubusercontent.com/42643563/93237929-0a11c500-f781-11ea-8a98-9402afce9d36.png)
> not develop branch as marked on that thread.

It's related to how composer manages PSR-4 and affects all versions. Since it's not a regression we add it for the 1.7.8 :wink:

**Comment by PierreRambaud** [comment]
I reopen since we need to update some dependencies :)

**Comment by LouiseBonnard** [comment]
Hi @PierreRambaud! Do you still need to update dependencies or this issue can be closed?

**Comment by PierreRambaud** [comment]
Still needed :)

**Comment by PierreRambaud** [comment]
I create a new pull request fixing one problem. Buf for the following one we will not do it because it's related to our dependencies:
` ` ` 
Class Sensio\Bundle\FrameworkExtraBundle\Tests\Request\ParamConverter\ArgumentNameConverterTest located in ./vendor/sensio/framework-extra-bundle/Tests/Request/ArgumentNameConverterTest.php does not comply with psr-4 autoloading standard. Skipping.
Class Sensio\Bundle\FrameworkExtraBundle\Tests\DependencyInjection\AddExpressionLanguageProvidersPassTest located in ./vendor/sensio/framework-extra-bundle/Tests/DependencyInjection/Compiler/AddExpressionLanguageProvidersPassTest.php does not comply with psr-4 autoloading standard. Skipping.
Class Sensio\Bundle\FrameworkExtraBundle\Tests\DependencyInjection\AddParamConverterPassTest located in ./vendor/sensio/framework-extra-bundle/Tests/DependencyInjection/Compiler/AddParamConverterPassTest.php does not comply with psr-4 autoloading standard. Skipping.
` ` ` 

Can be closed when #23210 is merged
